### ISSUE TITLE
NoDocOverPublicClass always reported for objects

### ISSUE THREAD

**Issue by sschuberth** [arturbosch/detekt#104]
For code like

    /**
     * Class docs not being recognized.
     */
    object Main {
        /**
         * The entry point for the application.
         *
         * @param args The list of process arguments.
         */
        @JvmStatic
        fun main(args: Array<String>) {
        }
    }

I always see `NoDocOverPublicClass - [Main] at src\main\kotlin\a.kt:1:1` being reported despite the docs being obviously present.

**Comment by arturbosch** [comment]
Thx for reporting!

**Comment by sschuberth** [comment]
@arturbosch can it be that `NoDocOverPublicClass` / `UndocumentedPublicClass` is now not reported at all anymore for `object`s, not even if docs are actually missing?

**Comment by arturbosch** [comment]
A minimal test case shows me that it works:

```kotlin
it("should report missing doc over object decl") {
    assertThat(subject.lint(""" object o { } """)).hasSize(1)
}
```

will try to find a real world object decl.

**Comment by arturbosch** [comment]
Or do you mean like a companion object inside a class ? That are not reported. The question is should they?

**Comment by sschuberth** [comment]
I was just using the example from my initial post but without the class docs, however the problem probably is on my side: I was using a non-standard profile name `all` like

```
detekt {
    profile('all') {
        config = 'detekt.yml'
    }
}
```

without specifying the non-standard profile name, so my settings in `detekt.yml` were ignored. However, if I use the standard `main` name like

```
detekt {
    profile('main') {
        config = 'detekt.yml'
    }
}
```

I get

    Provided path 'detekt.yml,detekt.yml' does not exist!

That used to work before. Where does the double `detekt.yml,detekt.yml` come form now?

**Comment by arturbosch** [comment]
Oh shit, that's the thing I forgot to remove yesterday. The main profile is considered two times that's why it tells you `detekt.yml,detekt.yml` but nevertheless it should work, if you would use an absolute path not just `detekt.yml` ? 

Something like `config = "$project.projectDir/detekt-cli/src/main/resources/default-detekt-config.yml"`

**Comment by sschuberth** [comment]
Still does not work with an absolute path, at least not on Windows:

```
:detektCheckException in thread "main" java.nio.file.InvalidPathException: Illegal char <:> at index 51:
C:\Users\sebastian\Development\project/detekt.yml,C:\Users\sebastian\Development\project/detekt.yml
```

**Comment by arturbosch** [comment]
I mean try using `$project.projectDir/detekt.yml` not absolute path - my fault never absolute path or CI will kill you :)

**Comment by sschuberth** [comment]
`${project.projectDir}/detekt.yml` is what I was using, which resolves to an absolute path at build time. Of course I did not *hard-code* an absolute path 😉

**Comment by arturbosch** [comment]
Ok so your on Windows and using `${project.projectDir}/detekt.yml` which results in 
```
:detektCheckException in thread "main" java.nio.file.InvalidPathException: Illegal char <:> at index 51:
C:\Users\sebastian\Development\project/detekt.yml,C:\Users\sebastian\Development\project/detekt.yml
``` correct?

Mh try `config = "${new File(project.projectDir, "detekt.yml")}"` and tell me what happens ... too bad I do not have Windows nearby

**Comment by sschuberth** [comment]
Yes, correct. Using your code results in the same error, but with the last `/` changed to a `\`, i.e. `C:\Users\sebastian\Development\project\detekt.yml`. That's it.

Maybe consider using AppVeyor CI for testing on Windows?

And: Should I file a now issue for this?

**Comment by arturbosch** [comment]
Yes please open a new issue.
